launcher/src/main/resources/com/threerings/getdown/launcher/PacUtils.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * Extracted from Mozilla Firefox 3.0.4, \mozilla\netwerk\base\src\nsProxyAutoConfig.js
+ */
+function dnsDomainIs(host, domain){
+	return (host.length >= domain.length &&
+		host.substring(host.length - domain.length) == domain);
+}
+
+function dnsDomainLevels(host){
+	return host.split('.').length-1;
+}
+
+function convert_addr(ipchars){
+	var bytes = ipchars.split('.');
+	var result = ((bytes[0] & 0xff) << 24) |
+		((bytes[1] & 0xff) << 16) |
+		((bytes[2] & 0xff) <<  8) |
+		(bytes[3] & 0xff);
+	return result;
+}
+
+function isInNet(ipaddr, pattern, maskstr){
+	var test = /^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})$/(ipaddr);

```

Comment:
This  line causes following exception in our environment:

javax.script.ScriptException: TypeError: /^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})$/ is not a function in <eval> at line number 23
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.throwAsScriptException(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.invokeImpl(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.NashornScriptEngine.invokeFunction(Unknown Source)
	at com.threerings.getdown.launcher.ProxyUtil.findProxyForURL(ProxyUtil.java:187)
	at com.threerings.getdown.launcher.ProxyUtil.autoDetectProxy(ProxyUtil.java:133)
	at com.threerings.getdown.launcher.Getdown.detectProxy(Getdown.java:200)
	at com.threerings.getdown.launcher.Getdown.run(Getdown.java:143)
Caused by: <eval>:23 TypeError: /^(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})\\.(\\d{1,3})$/ is not a function
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.error(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.typeError(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.typeError(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ECMAErrors.typeError(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptObject.notAFunction(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptObject.findCallMethod(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptObject.lookup(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.linker.NashornLinker.getGuardedInvocation(Unknown Source)
	at jdk.dynalink/jdk.dynalink.linker.support.CompositeTypeBasedGuardingDynamicLinker.getGuardedInvocation(Unknown Source)
	at jdk.dynalink/jdk.dynalink.linker.support.CompositeGuardingDynamicLinker.getGuardedInvocation(Unknown Source)
	at jdk.dynalink/jdk.dynalink.LinkerServicesImpl.lambda$getGuardedInvocation$0(Unknown Source)
	at jdk.dynalink/jdk.dynalink.LinkerServicesImpl.getWithLookupInternal(Unknown Source)
	at jdk.dynalink/jdk.dynalink.LinkerServicesImpl.getGuardedInvocation(Unknown Source)
	at jdk.dynalink/jdk.dynalink.DynamicLinker.relink(Unknown Source)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$10$531AAA$\^eval\_.isInNet(<eval>:23)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$4$36AA$\^eval\_.:scopeCall(<eval>)
	at jdk.scripting.nashorn.scripts/jdk.nashorn.internal.scripts.Script$Recompilation$4$36AA$\^eval\_.FindProxyForURL(<eval>:40)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptFunction.invoke(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.internal.runtime.ScriptRuntime.apply(Unknown Source)
	at jdk.scripting.nashorn/jdk.nashorn.api.scripting.ScriptObjectMirror.callMember(Unknown Source)
	... 6 more

Reply:
And that a call to function exec is missing:
var test = /^(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})$/.exec(ipaddr);